includes/gateway/class-omise-payment-installment.php

```diff
@@ -0,0 +1,262 @@
+					'description' => __( 'This controls the description which the user sees during checkout.', 'omise' )
+				),
+			);
+		}
+
+		/**
+		 * @inheritdoc
+		 */
+		public function payment_fields() {
+			$provider_names = array(
+				'installment_bay'          => __( 'Krungsri', 'omise' ),
+				'installment_first_choice' => __( 'Krungsri First Choice', 'omise' ),
+				'installment_kbank'        => __( 'Kasikorn Bank', 'omise' ),
+				'installment_bbl'          => __( 'Bangkok Bank', 'omise' ),
+				'installment_ktc'          => __( 'Krungthai Card (KTC)', 'omise' ),
+			);
+
+			$currency             = get_woocommerce_currency();
+			$cart_total           = WC()->cart->total;
+			$capabilities         = Omise_Capabilities::retrieve();
+			$installment_backends = $capabilities->getInstallmentBackends( $currency, $this->format_amount_subunit( $cart_total, $currency ) );
+
+			foreach ( $installment_backends as &$backend ) {
+				$backend->provider_code = str_replace( 'installment_', '', $backend->_id );
+				$backend->provider_name = isset( $provider_names[ $backend->_id ] ) ? $provider_names[ $backend->_id ] : strtoupper( $backend->provider_code );
+			}
+
+			Omise_Util::render_view(
+				'templates/payment/form-installment.php',
+				array( 'installment_backends' => $installment_backends )
+			);
+		}
+
+		/**
+		 * Register all required css and javascripts.
+		 */
+		public function omise_assets() {
+			if ( is_checkout() && $this->is_available() ) {
+				wp_enqueue_style( 'omise-css', plugins_url( '../../assets/css/omise-css.css', __FILE__ ), array(), OMISE_WOOCOMMERCE_PLUGIN_VERSION );
+			}
+		}
+
+		/**
+		 * @inheritdoc
+		 */
+		public function charge( $order_id, $order ) {
+			$source_type       = isset( $_POST['source']['type'] ) ? $_POST['source']['type'] : '';
+			$installment_terms = isset( $_POST[ $source_type . '_installment_terms'] ) ? $_POST[ $source_type . '_installment_terms'] : '';
+			$metadata          = array_merge(
+				apply_filters( 'omise_charge_params_metadata', array(), $order ),
+				array( 'order_id' => $order_id ) // override order_id as a reference for webhook handlers.
+			);
+
+			return OmiseCharge::create( array(
+				'amount'            => $this->format_amount_subunit( $order->get_total(), $order->get_order_currency() ),
+				'currency'          => $order->get_order_currency(),
+				'description'       => apply_filters( 'omise_charge_params_description', 'WooCommerce Order id ' . $order_id, $order ),
+				'source'            => array(
+					'type'              => sanitize_text_field( $source_type ),
+					'installment_terms' => sanitize_text_field( $installment_terms )
+				),
+				'return_uri'        => add_query_arg( 'order_id', $order_id, site_url() . '?wc-api=omise_installment_callback' ),
+				'metadata'          => $metadata
+			) );
+		}
+
+		/**
+		 * @inheritdoc
+		 */
+		public function result( $order_id, $order, $charge ) {
+			if ( self::STATUS_FAILED == $charge['status'] ) {
```

Comment:
Curious why you use this 'backwards' style for equality - it's not wrong, but it just doesn't read very well

Reply:
@jonrandy WordPress Code style guide, they called "Yoda style" 😂

Reply:
Ok. Odd style this is, but understand I do